platform/pensando/sonic-platform-modules-dpu/sonic_platform/watchdog.py

```diff
@@ -0,0 +1,50 @@
+# {C} Copyright 2023 AMD Systems Inc. All rights reserved
```

Comment:
@shanshri is there any specific reason why watchdog implementation is dummy? SONiC enables watchdog during reboot as part of firmware upgrade on sonic host/DPU so that if there is any issue in reboot, the DPU host does not remain stuck. Can we make use mainline watchdog kernel module? What is the DPU's watchdog driver?

Reply:
@prgeor, Internally our Pensando firmware container image have both software watchdog and hardware watchdog for same exact reason. There if any kernel panic or DPU stucks for some time then hardware watchdog kicks in. It is part of our bl31 and tested. We use following standard drivers.
```
root@sonic:/home/admin# lsmod | grep -e wdt -e soft
softdog                16384  3
dw_wdt                 16384  0
```

Reply:
@shanshri can you please enable the watchdog functionality in SONiC ie. use the sonic common watchdog functionality ?

Reply:
This will need further investigation and testing effort. We can push this to phase2.